File: javascript/hamming/hamming.js
function Hamming() {

    this.compute = function (source, compare) {
        if (source.length != compare.length) {
            throw new Error('DNA strands must be of equal length.');
        }
        var difference = 0;
        for (var i = 0; i < source.length; i++) {
            if (source[i] != compare[i])
                difference++;
        }
        return difference;
    }
}

module.exports = Hamming;

File: csharp/hamming/Hamming.cs
class Hamming
{
  public static int Compute(string source, string comparer)
  {
    var distance = 0;
    if (source.Length != comparer.Length)
    {
      return distance;
    }
    for (var i = 0; i < source.Length; ++i)
    {
      if (comparer[i] != source[i])
      {
        distance++;
      }
    }
    return distance;
  }
}